internal/app/go-filecoin/connectors/storage_market/client.go

```diff
@@ -229,5 +239,50 @@ func (s *StorageClientNodeConnector) GetChainHead(ctx context.Context) (shared.T
 }
 
 func (s *StorageClientNodeConnector) OnDealSectorCommitted(ctx context.Context, provider address.Address, dealID abi.DealID, cb storagemarket.DealSectorCommittedCallback) error {
-	panic("implement me") // @laser to do this
+	view, err := s.chainStore.StateView(s.chainStore.Head())
+	if err != nil {
+		cb(err)
+		return err
+	}
+
+	resolvedProvider, err := view.InitResolveAddress(ctx, provider)
+	if err != nil {
+		cb(err)
+		return err
+	}
+
+	err = s.waiter.WaitPredicate(ctx, func(msg *types.SignedMessage, c cid.Cid) bool {
+		if msg.Message.To != provider {
+			return false
+		}
+
+		// prove commit is method 8
+		if msg.Message.Method != abi.MethodNum(8) {
```

Comment:
Blocking: Eeek, this is bound to rot. Use the constant defined in `builtin`.

Reply:
replaced with builtin constant.